qubes/api/internal.py

```diff
@@ -361,3 +361,65 @@ async def suspend_post(self):
                         qubes.config.suspend_timeout,
                         "qubes.SuspendPostAll",
                     )
+
+    @qubes.api.method(
+        "internal.dispvm.preload.Create", scope="local", write=True
+    )
+    async def dispvm_preload_create(self):
+        """
+        Method called to create a preloaded DispVM.
+
+        :return:
+        """
+        dispvm_template = self.arg
+        preload_dispvm = dispvm_template.features.get("preload-dispvm", None)
+        ## TODO: should this function receive disposable name as argument
+        ##       or call admin.vm.CreateDisposable?
+        dispvm = TODO
+        if preload_dispvm:
+            preload_dispvm += " " + dispvm
+        else:
+            preload_dispvm = dispvm
+        dispvm_template.features["preload-dispvm"] = preload_dispvm
+        dispvm.features["internal"] = True
+
+    @qubes.api.method(
+        "internal.dispvm.preload.Use", scope="local", write=True
+    )
+    async def dispvm_preload_use(self):
```

Comment:
This should probably be an utility function somewhere, not an API method exposed to external callers.